deepchecks/base/check_context.py

```diff
@@ -0,0 +1,181 @@
+        return self.train.label_name
+
+    @property
+    def features(self):
+        if not self.train.features_columns:
+            raise DatasetValidationError('Check is irrelevant for Datasets without features')
+        return self.train.features
+
+    @property
+    def date_name(self):
+        if self.train.datetime_name is None:
+            raise DatasetValidationError('Check is irrelevant for Datasets without datetime column')
+        return self.train.datetime_name
+
+    @property
+    def index(self):
+        if self.train.index_name is None:
+            raise DatasetValidationError('Check is irrelevant for Datasets without an index')
+        return self.train.index_name
+
+    @property
+    def task_type(self):
+        if self._task_type is None:
+            self._task_type = task_type_check(self.model, self.train)
+        return self._task_type
+
+    @property
+    def features_importance(self):
+        if self._features_importance is None:
+            if self._feature_importance_mode == 'model':
+                self._features_importance = 0  # TODO get feature importance from model
+            elif self._feature_importance_mode == 'permutation':
+                self._features_importance = 0  # TODO get feature importance permutation
+            else:
+                raise DatasetValidationError(f'Unsupported feature importance mode: {self._feature_importance_mode}')
+        return self._features_importance
+
+    def assert_task_type(self, expected_types: Sequence[ModelType]):
+        if self.task_type not in expected_types:
+            raise ModelValidationError(
+                f'Check is relevant for models of type {[e.value.lower() for e in expected_types]}, '
+                f"but received model of type '{self.task_type.value.lower()}'"  # pylint: disable=inconsistent-quotes
+            )
+
+    def get_scorers(self, alternative_scorers: Mapping[str, Callable] = None, multiclass_non_avg=False):
+        if multiclass_non_avg:
+            scorers = alternative_scorers or self._user_non_avg_scorers
+        else:
+            scorers = alternative_scorers or self._user_scorers
+        return get_scorers_or_default(self.task_type, self.model, self.train, scorers, multiclass_non_avg)
+
+    def get_single_scorer(self, alternative_scorer: Union[str, Callable], multiclass_non_avg=False):
+        if alternative_scorer:
+            scorer = alternative_scorer
+        else:
+            user_scorers = self._user_non_avg_scorers if multiclass_non_avg else self._user_scorers
+            if user_scorers:
+                scorer_name = next(iter(self._user_non_avg_scorers))
+                scorer = self._user_non_avg_scorers[scorer_name]
+            else:
+                scorer = None
+        return get_single_scorer_or_default(self.task_type, self.model, self.train, scorer, multiclass_non_avg)
+
+
+class SingleDatasetBaseCheck(BaseCheck):
```

Comment:
Why are the subclasses of check inside a module called context?

Reply:
This is just temporary easier for me to update